src/annotator/anchoring/types.js

```diff
@@ -0,0 +1,210 @@
+      startOffset: selector.startOffset,
+      end: selector.endContainer,
+      endOffset: selector.endOffset,
+    };
+    const range = new SerializedRange(data);
+    return new RangeAnchor(root, range);
+  }
+
+  toRange() {
+    return this.range.toRange();
+  }
+
+  /**
+   * @param {Object} [options]
+   *   @param {string} [options.ignoreSelector]
+   * @return {RangeSelector}
+   */
+  toSelector(options = {}) {
+    const range = this.range.serialize(this.root, options.ignoreSelector);
+    return {
+      type: 'RangeSelector',
+      startContainer: range.start,
+      startOffset: range.startOffset,
+      endContainer: range.end,
+      endOffset: range.endOffset,
+    };
+  }
+}
+
+/**
+ * Converts between TextPositionSelector selectors and Range objects.
+ */
+export class TextPositionAnchor {
+  /**
+   * @param {Node|TextContentNode} root
+   * @param {number} start
+   * @param {number} end
+   */
+  constructor(root, start, end) {
+    this.root = root;
+    this.start = start;
+    this.end = end;
+  }
+
+  /**
+   * @param {Node} root
+   * @param {Range} range
+   */
+  static fromRange(root, range) {
+    const selector = posFromRange(root, range);
+    return TextPositionAnchor.fromSelector(root, selector);
+  }
+  /**
+   * @param {Node} root
+   * @param {TextPositionSelector} selector
+   */
+  static fromSelector(root, selector) {
+    return new TextPositionAnchor(root, selector.start, selector.end);
+  }
+
+  /**
+   * @return {TextPositionSelector}
+   */
+  toSelector() {
+    return {
+      type: 'TextPositionSelector',
+      start: this.start,
+      end: this.end,
+    };
+  }
+
+  toRange() {
+    return posToRange(this.root, { start: this.start, end: this.end });
+  }
+}
+
+/**
+ * Converts between TextQuoteSelector selectors and Range objects.
+ */
+export class TextQuoteAnchor {
+  /**
+   * @param {Node|TextContentNode} root - A root element from which to anchor.
+   * @param {string} exact
+   * @param {Object} context
+   *   @param {string} [context.prefix]
+   *   @param {string} [context.suffix]
+   */
+  constructor(root, exact, context = {}) {
+    this.root = root;
+    this.exact = exact;
+    this.context = context;
+  }
+  /**
+   * @param {Node} root
+   * @param {Range} range
+   */
+  static fromRange(root, range) {
+    const selector = quoteFromRange(root, range);
+    return TextQuoteAnchor.fromSelector(root, selector);
+  }
+
+  /**
+   * @param {Node|TextContentNode} root
+   * @param {TextQuoteSelector} selector
+   */
+  static fromSelector(root, selector) {
+    const { prefix, suffix } = selector;
+    return new TextQuoteAnchor(root, selector.exact, { prefix, suffix });
+  }
+
+  /**
+   * @return {TextQuoteSelector}
+   */
+  toSelector() {
+    return {
+      type: 'TextQuoteSelector',
+      exact: this.exact,
+      prefix: this.context.prefix,
+      suffix: this.context.suffix,
+    };
+  }
+
+  /**
+   * @param {Object} [options]
+   *   @param {Object} [options.hint]
```

Comment:
A comment here would be useful to clarify what `hint` is. This can point to relevant part of the dom-anchor-text-quote README if there is one.